scripts/breaking_changes_checker/changelog_tracker.py

```diff
@@ -46,6 +46,19 @@ def __init__(self, stable: Dict, current: Dict, diff: Dict, package_name: str, *
     def run_checks(self) -> None:
         self.run_non_breaking_change_diff_checks()
         super().run_checks()
+        self.run_async_changelog_cleanup()
+
+    # Remove duplicate reporting of changes that apply to both sync and async package components
+    def run_async_changelog_cleanup(self) -> None:
+        # Create a list of all sync changes
+        non_aio_changes = [fa for fa in self.features_added if "aio" not in fa[2]]
+        # Remove any aio change if there is a sync change that is the same
+        for change in non_aio_changes:
+            for c in self.features_added:
+                if "aio" in c[2]:
+                    if change[1] == c[1] and change[3:] == c[3:]:
+                        self.features_added.remove(c)
+                        break
```

Comment:
It seems duplicated code and it is better to extract a separated function.

Reply:
Done,thanks!